src/codeflare_sdk/cluster/widgets.py

```diff
@@ -89,3 +100,234 @@ def is_notebook() -> bool:
+            plural = "rayclusters"
+
+        # Wait for the resource to be deleted
+        while timeout > 0:
+            try:
+                api_instance.get_namespaced_custom_object(
+                    group=group,
+                    version=version,
+                    namespace=namespace,
+                    plural=plural,
+                    name=cluster_name,
+                )
+                # Retry if resource still exists
+                time.sleep(interval)
+                timeout -= interval
+                if timeout <= 0:
+                    raise TimeoutError(f"Timeout waiting for {cluster_name} to be deleted.")
+            except ApiException as e:
+                # Resource is deleted
+                if e.status == 404:
+                    break
+    except Exception as e:
+        return _kube_api_error_handling(e)
+
+
+def _fetch_cluster_data(namespace):
+    from .cluster import list_all_clusters
+    rayclusters = list_all_clusters(namespace, False)
+    if not rayclusters:
+        return pd.DataFrame()
+    names = [item.name for item in rayclusters]
+    namespaces = [item.namespace for item in rayclusters]
+    head_extended_resources = [
+        f"{list(item.head_extended_resources.keys())[0]}: {list(item.head_extended_resources.values())[0]}"
+        if item.head_extended_resources else "0"
+        for item in rayclusters
+    ]
+    worker_extended_resources = [
+        f"{list(item.worker_extended_resources.keys())[0]}: {list(item.worker_extended_resources.values())[0]}"
+        if item.worker_extended_resources else "0"
+        for item in rayclusters
+    ]
+    head_cpu_requests = [item.head_cpu_requests if item.head_cpu_requests else 0 for item in rayclusters]
+    head_cpu_limits = [item.head_cpu_limits if item.head_cpu_limits else 0 for item in rayclusters]
+    head_cpu_rl = [f"{requests}~{limits}" for requests, limits in zip(head_cpu_requests, head_cpu_limits)]
+    head_mem_requests = [item.head_mem_requests if item.head_mem_requests else 0 for item in rayclusters]
+    head_mem_limits = [item.head_mem_limits if item.head_mem_limits else 0 for item in rayclusters]
+    head_mem_rl = [f"{requests}~{limits}" for requests, limits in zip(head_mem_requests, head_mem_limits)]
+    worker_cpu_requests = [item.worker_cpu_requests if item.worker_cpu_requests else 0 for item in rayclusters]
+    worker_cpu_limits = [item.worker_cpu_limits if item.worker_cpu_limits else 0 for item in rayclusters]
+    worker_cpu_rl = [f"{requests}~{limits}" for requests, limits in zip(worker_cpu_requests, worker_cpu_limits)]
+    worker_mem_requests = [item.worker_mem_requests if item.worker_mem_requests else 0 for item in rayclusters]
+    worker_mem_limits = [item.worker_mem_limits if item.worker_mem_limits else 0 for item in rayclusters]
+    worker_mem_rl = [f"{requests}~{limits}" for requests, limits in zip(worker_mem_requests, worker_mem_limits)]
+    status = [item.status.name for item in rayclusters]
```

Comment:
I think it would be beneficial to include `num_workers` as things can be confusing when GPUs are involved see below:
![image](https://github.com/user-attachments/assets/ed6fc443-5865-48f9-8b2c-2e8e273c133d)
If I didn't know how many workers were on this Ray Cluster I wouldn't know how many GPUs in total I am using either.

Reply:
Good call!

Reply:
In near future, we could look to implement the status of head and workers too as described in the refinement document.